lib/component.js

```diff
@@ -314,14 +314,15 @@ class Component extends WebComponent {
     if (this.router) {
       this.router.unregisterListeners();
     }
-    if (this.$panelParent) {
-      this.$panelParent.$panelChildren.delete(this);
```

Comment:
i think removing this line will lead to bad tree behavior. if this change isn't causing tests to fail then we probably need some tests verifying that removing a panel component from the DOM removes it from the panel tree too